Controller/DemoController.php

```diff
@@ -21,6 +21,130 @@
 class DemoController extends Controller
 {
     /**
+     * Displays the sub folder if it exists
+     *
+     * @param int $locationId of a folder
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function showSubFolderLinkAction( $locationId )
+    {
+        $response = new Response();
+
+        $repository = $this->getRepository();
+        $location = $repository->getLocationService()->loadLocation( $locationId );
+        if ( $location->invisible )
+        {
+            throw new NotFoundHttpException( "Location #$locationId cannot be displayed as it is flagged as invisible." );
+        }
+
+        $languages = $this->getConfigResolver()->getParameter( 'languages' );
+
+        $subContentCriteria = $this->get( 'ezdemo.criteria_helper' )->generateSubContentCriterion(
+            $location, $languages
+        );
+
+        $subContentQuery = new Query();
+        $subContentQuery->criterion = $subContentCriteria;
+        $subContentQuery->sortClauses = array(
+            new SortClause\ContentName()
+        );
+
+        $searchService = $this->getRepository()->getSearchService();
+        $subContent = $searchService->findContent( $subContentQuery );
+
+        foreach ( $subContent->searchHits as $hit )
+        {
+            $subFolders[] = $hit->valueObject;
+        }
+
+        return $this->render(
+            'eZDemoBundle:aside:folder.html.twig',
+            array(
+                'location' => $location,
+                'subFolders' => $subFolders
+            ),
+            $response
```

Comment:
as we have defined a custom controller for this, shouldn't we specificy ```SharedMaxAge``` here for caching?